apps/server/src/modules/h5p-editor/service/contentStorage.service.ts

```diff
@@ -23,12 +24,22 @@ export class ContentStorage implements IContentStorage {
 		@Inject('S3ClientAdapter_Content') private readonly storageClient: S3ClientAdapter
 	) {}
 
+	private checkExtendedUserType(user: IUser) {
+		const isExtendedUserType = user instanceof LumiUserWithContentData;
+
+		if (!isExtendedUserType) {
+			throw new Error();
+		}
+	}
+
 	public async addContent(
 		metadata: IContentMetadata,
 		content: unknown,
-		user: IUser,
+		user: LumiUserWithContentData,
 		contentId?: ContentId | undefined
 	): Promise<ContentId> {
+		this.checkExtendedUserType(user);
```

Comment:
Should this be moved to the "try" section?

Reply:
Moved inside try so error gets handled by the method